assembly/__tests__/safe_u128.spec.as.ts

```diff
@@ -0,0 +1,302 @@
+    var a = u128.Max;
+    var b = u128.Zero;
+    expect<u128>(a + b).toStrictEqual(u128.Max);
+  });
+
+  it("should Add Two Numbers 3", (): void => {
+    var a = u128.from(-2);
+    var b = u128.One;
+    expect<u128>(a + b).toStrictEqual(u128.Max);
+  });
+
+  it("should Subtract Two Numbers 1", (): void => {
+    var a = new u128(355, 355);
+    var b = new u128(100, 255);
+    expect<u128>(a - b).toStrictEqual(new u128(255, 100));
+  });
+
+  it("should Subtract Two Numbers 2", (): void => {
+    var a = u128.Max;
+    var b = u128.Max;
+    expect<u128>(a - b).toStrictEqual(u128.Zero);
+  });
+
+  it("should Subtract Two Numbers 3", (): void => {
+    var a = u128.Max;
+    var b = u128.Zero;
+    expect<u128>(a - b).toStrictEqual(u128.Max);
+  });
+
+  it("should Multiply Two Numbers 1", (): void => {
+    var a = u128.from(43545453452);
+    var b = u128.Zero;
+    expect<u128>(a * b).toStrictEqual(u128.Zero);
+  });
+
+  it("should Multiply Two Numbers 2", (): void => {
+    var a = u128.from(43545453452);
+    var b = u128.One;
+    expect<u128>(a * b).toStrictEqual(a);
+  });
+
+  it("should Multiply Two Numbers 3", (): void => {
+    var a = u128.Max;
+    var b = u128.One;
+    expect<u128>(a * b).toStrictEqual(a);
+  });
+
+  it("should Multiply Two Numbers 4", (): void => {
+    var a = u128.from(43545453452);
+    var b = u128.from(2353454354);
+    expect<u128>(a * b).toStrictEqual(new u128(10248516654965971928, 5));
+  });
+
+  it("should Power Number 1", (): void => {
+    expect<u128>(u128.Zero ** 1024).toStrictEqual(u128.Zero);
+  });
+
+  it("should Power Number 2", (): void => {
+    expect<u128>(u128.One ** 0).toStrictEqual(u128.One);
+  });
+
+  it("should Power Number 3", (): void => {
+    expect<u128>(u128.One ** 1024).toStrictEqual(u128.One);
+  });
+
+  it("should Power Number 4", (): void => {
+    expect<u128>(new u128(2) ** 127).toStrictEqual(new u128(0, 0x8000000000000000));
+  });
+
+  it("should Power Number 5", (): void => {
+    expect<u128>(new u128(3) ** 80).toStrictEqual(new u128(0x3CEA59789C79D441, 0x6F32F1EF8B18A2BC));
+  });
+
+  it("should Power Number 6", (): void => {
+    expect<u128>(new u128(4) ** 63).toStrictEqual(new u128(0, 0x4000000000000000));
+  });
+
+  it("should Power Number 7", (): void => {
+    expect<u128>(new u128(5) ** 55).toStrictEqual(new u128(0xFFF4B4E3F741CF6D, 0xD0CF4B50CFE20765));
+  });
+
+  it("should Power Number 8", (): void => {
+    expect<u128>(new u128(6) ** 49).toStrictEqual(new u128(0x4286000000000000, 0x6558E2A0921FE069));
+  });
+
+  it("should Power Number 9", (): void => {
+    expect<u128>(new u128(0, 1) ** 1).toStrictEqual(new u128(0, 1));
+  });
+
+  it("should Power Number 10", (): void => {
+    expect<u128>(new u128(u64.MAX_VALUE) ** 2).toStrictEqual(new u128(1, 0x0FFFFFFFFFFFFFFFE));
+  });
+})
+
+describe("Overflow Underflow Throwable", () => {
+  it("should Throw When Add Two Numbers 1", (): void => {
+    expectFn(()=>{
```

Comment:
Could you add spaces between `=>`?

Reply:
Sure, it was from a regex and by a regex it will be fixed.